seth/tts-CalculatorAssignment/JsKoo.js

```diff
@@ -0,0 +1,114 @@
+var onebtn = document.getElementById('calc-one');
+var onebtn = document.getElementById('calc-two');
+var onebtn = document.getElementById('calc-three');
+var onebtn = document.getElementById('calc-four');
+var onebtn = document.getElementById('calc-five');
+var onebtn = document.getElementById('calc-six');
+var onebtn = document.getElementById('calc-seven');
+var onebtn = document.getElementById('calc-eight');
+var onebtn = document.getElementById('calc-nine');
+var onebtn = document.getElementById('calc-zero');
+
+var decimalbtn = document.getElementById('decimal');
+var clearbtn = document.getElementById('clear');
+var backpacebtn = document.getElementById('backspace');
+var displayValElement = document.getElementById('Calculator-display');
+
+var displayVal = '0';
+var pendingVal;
+var evalStringArray = [];     /* problem here */
+
+var calcNumnbtns = document.getElementsByClassName('calc-btn-num');
+var calOperationBtns = document.getElementsByClassName('calc-btn-operator');
+
+var updateDisplayVal = (clickObj) => {
+  var btnText = clickObj.target.innerText;
+
+if(displayVal === '0')
+  displayVal = '';
+
+displayVal += btnText;
+displayValElement.innerText = displayVal;
+}
+
+var performOperation = (clickObj) => {
+  var operator = clickObj.target.innerText;
+
+    switch (operator) {
+      case '+':
+        pendingVal = displayVal;
+        display ='0';
+        displayValElement.innerText = displayVal;
+        evalStringArray.push(pendingVal);
+        evalStringArray.push('+');
+        break;
+
+      case '-':
+        pendingVal = displayVal;
+        display ='0';
+        displayValElement.innerText = displayVal;
+        evalStringArray.push(pendingVal);
+        evalStringArray.push('-');
+        break;
+
+        case 'x':
+          pendingVal = displayVal;
+          display ='0';
+          displayValElement.innerText = displayVal;
+          evalStringArray.push(pendingVal);
+          evalStringArray.push('*');
+          break;
+
+        case '÷':
+          pendingVal = displayVal;
+          display ='0';
+          displayValElement.innerText = displayVal;
+          evalStringArray.push(pendingVal);
+          evalStringArray.push('/');
+
+        case '=':
+          evalStringArray.push(displayVal);
+          var evaluation = eval(evalStringArray.join(''));
+          displayVal = evaluation + '';
+          displayValElement.innerText = displayVal;
+          evalStringArray = [];
+          break;
+      default:
+          break;
+
+    }
+
+}
+
+
+
+for (let i = 0; i < calcNumnbtns.length; i++) {
+  calcNumnbtns[i].addEventListener('click', updateDisplayVal, false);
+}
+for (let i = 0; i < calOperationBtns.length; i++) {
+  calOperationBtns[i].addEventListener('click', performOperation, false);
+}
+
+
+
+clearbtn.onclick = () => {
```

Comment:
Typically you wouldn't ever assign a var with `undefined`.  Undefined is used to show you whether or not a property has been assigned. You'd typically use `null` if you wanted to unassign it. Here though I'm not sure it makes a difference since you're setting and using pendingVal in your other functions. I think this function might work just as well like this: 

```javascript 
clearbtn.onclick = () => {
  displayValElement.innerHTML = "0";
}

```